src/main/java/io/javalin/http/ResponseHandler.kt

```diff
@@ -0,0 +1,69 @@
+package io.javalin.http
+
+import io.javalin.core.compression.CompressionStrategy
+import io.javalin.core.util.Header
+import javax.servlet.ServletOutputStream
+import javax.servlet.WriteListener
+import javax.servlet.http.HttpServletResponse
+import javax.servlet.http.HttpServletResponseWrapper
+
+/**
+ * This class acts as an "interceptor" for handling both static and dynamic responses.
+ * We use it to wrap the "res" parameter inside JavalinServlet#service
+ *
+ * For dynamic responses, we then simply call res.outputStream.write() inside service#writeResult and feed it the resultStream
+ *
+ * For static responses, we intercept Jetty's writing by feeding its ResourceHandler the wrapped "res" param instead of the original.
+ * Jetty uses the same outputStream.write() method to write static content to output after handling it.
+ * Because of this, we are able to capture Jetty's output in here, modify it as needed, then commit it ourselves
+ *
+ * For the moment, this is being used to bypass Jetty's compression handling, so we can use our own (which supports Brotli and dynamic content),
+ * however, there could be more uses for this in the future.
+ */
+class ResponseHandler(
```

Comment:
I think I would call this `JavalinResponseWrapper`, and try to move more parts of `writeResult` in here.